src/PostgREST/AppState.hs

```diff
@@ -515,6 +516,7 @@ listener appState@AppState{stateObserver=observer} conf@AppConfig{..} = do
       putIsListenerOn appState False
       -- assume the pool connection was also lost, call the connection worker
       connectionWorker appState
+      -- TODO: debounce
       -- retry the listener
       listener appState conf
```

Comment:
This needs a debounce for when the connection is lost or the `db-channel-uri` has an error.

Since this should apply only when `db-channel-uri` is set, perhaps it should run conditionally, like:

```suggestion
      when (isNothing dbChannel) $
        connectionWorker appState
      -- TODO: debounce
      -- retry the listener
      listener appState conf
```